### PR TITLE
adding `updateHorse`, `registerIHOHorse`, `closeIHOHorse`, and `updateHorseTrainingFee` transactions

### PR DESCRIPTION
**Pull request by capt4ce**
## Description

adding `updateHorse`, `registerIHOHorse`, `closeIHOHorse`, and `updateHorseTrainingFee` transactions.

## Breakdown

- [x] adding the transaction constants in the `packages/keyhub-vault-nxt/lib/conf/constants.js`
- [x] adding the transaction constants in the `packages/keyhub-vault-nxt/lib/js/nrs.server.js`
- [x] adding the transaction constants in the `packages/keyhub-vault-web/src/conf/eqh/constants.js`
- [x] adding transaction validation form `updateHorse`, `registerIHOHorse`, `closeIHOHorse`, and `updateHorseTrainingFee` transactions

## Step to Test (optional)

- registerIHOHorse
```
const tx = {
      type: `registerIHOHorse`,
      data: {
        horseId: 5,
        type: 'listed',
        name: 'horse5',
        birthDate: '2004-12-02',
        sireId: 5,
        sireName: 'anything',
        damId: 6,
        damName: 'not anything',
        trainingFee: 800,
        ihoQuantity: 20000,
        ihoPrice: '120',
        startTimestamp: 5,
        endTimestamp: 10,
        lotSize: 150,
        requiredMonthDeposit: 3,
        quantityQNT: 100000,
      },
    }
```
- closeIHOHorse
```
const tx = {
      type: `closeIHOHorse`,
      data: {
        horseId: 5,
      },
    }
```

- updateHorseTraningFee
```
const tx = {
      type: `updateHorseTrainingFee`,
      data: {
        horseId: 5,
        trainingFee: 200,
        startTimestamp: 10,
      },
    }
```

### CODE REVIEW

File: packages/keyhub-vault-nxt/lib/conf/constants.js [hunk 1]
```diff
@@ -1734,6 +1734,38 @@ NRS.constants.SERVER = {
...
+      enabled: true,
+    },
+    updateHorseTrainingFee: {
+      allowRequiredBlockParameters: false,
+      requireFullClient: false,
+      requirePassword: false,
+      requireBlockchain: true,
+      requirePost: true,
+      enabled: true,
+    },
```
Comment (heri16) [inline packages/keyhub-vault-nxt/lib/conf/constants.js]: Theses transactions do not need to be in this file as they are EQH specific.

File: packages/keyhub-vault-nxt/lib/js/nrs.server.js [hunk 2]
```diff
@@ -1342,6 +1342,26 @@ var NRS = (function (NRS, $, undefined) {
...
+            case "closeIHOHorse":
+                if (transaction.type !== 8 || transaction.subtype !== 2) {
+                    return false;
+                }
+                break;
+            case "updateHorseTrainingFee":
+                if (transaction.type !== 8 || transaction.subtype !== 3) {
+                    return false;
+                }
+                break;
```
Comment (heri16) [inline packages/keyhub-vault-nxt/lib/js/nrs.server.js]: Need to decode the transaction bytes of all custom fields in order to actually verify the content of the transaction.

Comment (heri16) [inline packages/keyhub-vault-nxt/lib/js/nrs.server.js]: Need to decode the transaction bytes of all custom fields in order to actually verify the content of the transaction.

Comment (heri16) [reply packages/keyhub-vault-nxt/lib/js/nrs.server.js]: Purpose of `NRS.verifyTransactionTypes()` is to decode the bytes using the established byteorder format of the custom transaction, then compare if all the fields are the same as the json data sent.

File: packages/keyhub-vault-nxt/lib/conf/constants.js (hunk 1 above)
Comment (capt4ce) [reply packages/keyhub-vault-nxt/lib/conf/constants.js]: removed

File: packages/keyhub-vault-nxt/lib/js/nrs.server.js (hunk 2 above)
Comment (capt4ce) [reply packages/keyhub-vault-nxt/lib/js/nrs.server.js]: done

**Pull request merged**